R/bootstrap-model.R

```diff
@@ -173,20 +165,11 @@ setup_bootstrap_run <- function(
 
     if(is.null(data)){
       # Only include subjects that entered the original problem by default
-      can_be_joined <- can_be_nm_joined(orig_mod, .join_col = .join_col)
-      if(isTRUE(can_be_joined)){
-        starting_data <- nm_join(orig_mod, .join_col = .join_col) %>%
-          suppressMessages()
-
-        # select only columns from original data set
-        starting_data <- starting_data %>%
-          dplyr::select(attr(starting_data, "nm_join_origin")$data)
-
-        if ("DV.DATA" %in% names(starting_data)) {
-          starting_data <- dplyr::rename(starting_data, "DV" = "DV.DATA")
-        }
-      }else{
-        rlang::inform(
+      starting_data <- nm_data(orig_mod, filter = TRUE) %>% suppressMessages()
```

Comment:
One point and two questions, though id like @seth127's opinion here too:

 point: 
 - It's my understanding that we would prefer to keep this away from `nm_data()`. I think `setup_bootstrap_run` would have to be the landing spot for this, but this is what i'd like Seth's opinion on.

questions:
 - I started doing this, but we'd need to add a `.bbi_args` argument like we do for `nm_join()` if we were to do that to avoid `model_summary()` issues. Does doing this seem out of place for `setup_bootstrap_run`?
 - Im unsure if this should be a `dev_error()` versus a regular abort call or even a warning. Are we adamant that if there's a mismatch, the filtering _definitely_ went wrong, or is it possible a mismatch could arise for a different reason?

Reply:
@kylebaron Im proposing a message like the one below, though let me know if it's too verbose.

 Im **assuming we'll want to error out here**, but wanted to pose the option of prompting the user if this is ok. My understanding is that diverging from the number of expected records would mean the bootstrap run is invalid. However since we allow users to manually adjust the bootstrap template control stream (after `new_bootstrap_run` is called), I was wondering if there could ever be an intentional deviation from the model the bootstrap is based on? If that's a possibility, I would propose something like this:
![image](https://github.com/user-attachments/assets/0d956ddb-fefd-4abb-836f-56cf22f479e3)